hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAsyncClusterAdminApi.java

```diff
@@ -65,19 +67,47 @@
   public static final HBaseClassTestRule CLASS_RULE =
       HBaseClassTestRule.forClass(TestAsyncClusterAdminApi.class);
 
-  private final Path cnfPath = FileSystems.getDefault().getPath("target/test-classes/hbase-site.xml");
-  private final Path cnf2Path = FileSystems.getDefault().getPath("target/test-classes/hbase-site2.xml");
-  private final Path cnf3Path = FileSystems.getDefault().getPath("target/test-classes/hbase-site3.xml");
+  private static Path newCnfPath, newCnf2Path, newCnf3Path;
 
   @BeforeClass
   public static void setUpBeforeClass() throws Exception {
+    Path cnfPath = FileSystems.getDefault().getPath("target/test-classes/hbase-site.xml");
+    Path cnf2Path = FileSystems.getDefault().getPath("target/test-classes/hbase-site2.xml");
+    Path cnf3Path = FileSystems.getDefault().getPath("target/test-classes/hbase-site3.xml");
+
+    String absoluteDataPath = TEST_UTIL.getDataTestDir().toString();
+    String dataBasePath = System.getProperty(HBaseCommonTestingUtility.BASE_TEST_DIRECTORY_KEY,
+      HBaseCommonTestingUtility.DEFAULT_BASE_TEST_DIRECTORY);
```

Comment:
This is interesting. Do you not trust what you get from TEST_UTIL#getDataTestDir?

Reply:
The one from getDataTestDir is the absolute path. It needs to get the relative path here so need to have base directory. Maybe it should not expose such details from internal, let me wrap up an API from TEST_UTIL.

Reply:
Hi @saintstack, I made HBaseCommonTestingUtility#getBaseTestDir public as I need this API to avoid hack the details of how base test dir is created in the testing code.

Reply:
Sounds good.